### PR TITLE
Scaladoc bugfixes for reflection

### PR DESCRIPTION
**Pull request by VladUreche**
Review by @vigdorchik @jsuereth

### CODE REVIEW

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 1]
```diff
@@ -31,6 +31,12 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
   import definitions.{ ObjectClass, NothingClass, AnyClass, AnyValClass, AnyRefClass }
   import rootMirror.{ RootPackage, RootClass, EmptyPackage }
 
+  // Defaults for member grouping, that may be overridden by the template
+  val defaultGroup = "Ungrouped"
```
Comment (jsuereth) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:35]: Changed from "No Group" to "Ungrouped".  Does that cause any issues, or just a UI thing?

Comment (VladUreche) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:35]: UI only

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 2]
```diff
@@ -286,7 +292,8 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
         val tps = (this match {
           case a: AliasType => sym.tpe.dealias.parents
           case a: AbstractType => sym.info.bounds match {
-            case TypeBounds(lo, hi) => List(hi)
+            case TypeBounds(lo, RefinedType(parents, decls)) => parents
```
Comment (jsuereth) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:295]: If this is doing what I think it's doing, then god bless on this change.   Reflection docs were really tough to navigate before this.

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 3]
```diff
@@ -493,16 +500,21 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
         if (entity != default) return entity
       }
       // query linearization
-      for (tpl <- linearizationTemplates.collect{ case dtpl: DocTemplateImpl if dtpl!=this => dtpl}) {
-        val entity = tpl.groupSearch(extractor, default)
-        if (entity != default) return entity
-      }
-      default
+      if (!sym.isPackage)
```
Comment (jsuereth) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:503]: This is ok for package objects?

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala (hunk 2 above)
Comment (VladUreche) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:295]: Shhhh! Don't spoil the surprise!

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala (hunk 3 above)
Comment (VladUreche) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:503]: In the scaladoc model package object contents are squashed into the package.

**Comment by vigdorchik** [comment]
LGTM.

**Comment by jsuereth** [comment]
Ok, this LGTM too, and others.   THIS IS THE LAST EXCEPTION WE MAKE.   I'm sorry, but no other code changes.  I've been far too lenient here.  Eveyrthing else on 2.10.1

**Pull request merged**